### PR TITLE
Memory optimisation

### PR DESCRIPTION
**Pull request by Bobronium**
### Motivation and context
By using tuple as a container for points when dealing with import from datumaro, we can achieve 2 things:
- Reduce memory needed for copying shapes and tracks during import (running `deepcopy` on `tuple[int]` will return the same object, as opposed to `list[int]`)
- Guarantee type safety during later stages of data pipeline and skip additional conversion added in #1898

Same thing arguable should be done for CVAT format as well.

Benchmarks: [memray_reports.zip](https://github.com/user-attachments/files/16849509/memray_reports.zip)

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Improved handling of shape points during the import process for enhanced data accuracy.
	- Centralized conversion of shape points to floats, optimizing memory usage and performance.

- **Refactor**
	- Enhanced code readability and maintainability by restructuring the point conversion logic.

### CODE REVIEW

File: cvat/apps/dataset_manager/bindings.py [hunk 1]
```diff
@@ -567,14 +563,27 @@ def _import_track(self, track, parent_label_id=None):
...
+        Historically, there were importers that were not converting points to ints/floats.
+        The only place to make sure that all points in shapes have the right type was this one.
+        However, this does eat up a lot of memory for some reason.
+        (see https://github.com/cvat-ai/cvat/pull/8226/files#r1695445137)
+
+        So, before we can guarantee that all the importers are returning the right data,
+        we have to have this conversion.
+        """
+        # if points is LazyList or tuple, we can be sure it has the right type already
+        if isinstance(points := shape["points"], LazyList | tuple):
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:581]: How a tuple guarantees the correct type? Wouldn't be better to go through the input and check types instead?

File: cvat/apps/dataset_manager/bindings.py [hunk 2]
```diff
@@ -5,10 +5,10 @@
 
 from __future__ import annotations
 
+import logging
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:8]: Probably, you should use the task logger or the global server logger instead. Defined in `engine/log.py`.

File: cvat/apps/dataset_manager/bindings.py [hunk 3]
```diff
@@ -188,20 +191,76 @@ def _export_attributes(self, attributes):
...
+        id: int | None = None
+
+    class TrackedShape(NamedTuple):
+        type: int = 0
+        frame: int = 0
+        points: Sequence[int] = ()
+        occluded: bool = False
+        outside: bool = False
+        keyframe: bool = False
+        attributes: Sequence[int] = ()
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:220]: I'm not sure the added defaults are useful, especially tuples. How do you think they should be used?

File: cvat/apps/dataset_manager/bindings.py (hunk 2 above)
Comment (Bobronium) [reply cvat/apps/dataset_manager/bindings.py]: Agreed, thanks.

File: cvat/apps/dataset_manager/bindings.py (hunk 3 above)
Comment (Bobronium) [reply cvat/apps/dataset_manager/bindings.py]: Regarding added defaults, you're right. I've removed ones that weren't there before.

As for tuples, they serve as a replacement for lists that were in place before. 

You never want to assign a mutable collection as a default attribute:

```py
class Test(NamedTuple):
    attr: list[str] = []


first_instance = Test()
second_instance = Test()

print(first_instance is not second_instance)  # True
print(first_instance.attr is second_instance.attr)  # True
first_instance.attr.append("foo")
print(first_instance, second_instance)  # Test(attr=['foo']) Test(attr=['foo'])
```

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**